numpy/_core/src/umath/loops_arithm_fp.dispatch.cpp

```diff
@@ -0,0 +1,766 @@
+    for(i = 0; i < n; i++, indxp += isindex, value += isb) {
+        npy_intp indx = *(npy_intp *)indxp;
+        if (indx < 0) {
+            indx += shape;
+        }
+        indexed  = (T *)(ip1 + is1 * indx);
+        *indexed = op_func(*indexed, *(T *)value);
+    }
+    return 0;
+}
+
+template <typename T, typename Op>
+HWY_INLINE HWY_ATTR void
+complex_single_double_func(char **args, npy_intp const *dimensions, npy_intp const *steps)
+{
+    Op op_func;
+    using D = std::conditional_t<sizeof(T) == 8, int64_t, int32_t>;
+    npy_intp len = dimensions[0];
+    char *b_src0 = args[0], *b_src1 = args[1], *b_dst = args[2];
+    npy_intp b_ssrc0 = steps[0], b_ssrc1 = steps[1], b_sdst = steps[2];
+    if constexpr (Op::PW) {
+        // reduce
+        if (b_ssrc0 == 0 && b_ssrc0 == b_sdst && b_src0 == b_dst &&
+            b_ssrc1 % (sizeof(T)*2) == 0
+        ) {
+            T *rl_im = (T *)b_src0;
+            T rr, ri;
+            if constexpr (std::is_same_v<T, float>){
+                CFLOAT_pairwise_sum(&rr, &ri, b_src1, len * 2, b_ssrc1 / 2);
+            } else {
+                CDOUBLE_pairwise_sum(&rr, &ri, b_src1, len * 2, b_ssrc1 / 2);
+            }
+            rl_im[0] = op_func(rl_im[0], rr);
+            rl_im[1] = op_func(rl_im[1], ri);
+            return;
+        }
+    }
+
+#if NPY_HWY
+    if constexpr (kSupportLane<T>) {
+        // Certain versions of Apple clang (commonly used in CI images) produce
+        // non-deterministic output in the mul path with AVX2 enabled on x86_64.
+        // Work around by scalarising.
+        #if defined(NPY_CPU_AMD64) && defined(__clang__) \
+                && defined(__apple_build_version__) \
+                && __apple_build_version__ >= 14000000 \
+                && __apple_build_version__ < 14030000
+            if constexpr (Op::is_mul) {
+                goto loop_scalar;
+            }
+        #endif  // end affected Apple clang.
+
+        if (!is_mem_overlap(b_src0, b_ssrc0, b_dst, b_sdst, len) &&
+            !is_mem_overlap(b_src1, b_ssrc1, b_dst, b_sdst, len) &&
+            sizeof(T) == alignof(T) && b_ssrc0 % sizeof(T) == 0 && b_ssrc1 % sizeof(T) == 0 && 
+            b_sdst % sizeof(T) == 0 && b_sdst != 0) {
+            const T *src0 = (T*)b_src0;
+            const T *src1 = (T*)b_src1;
+                  T *dst  = (T*)b_dst;
+
+            const npy_intp ssrc0 = b_ssrc0 / sizeof(T);
+            const npy_intp ssrc1 = b_ssrc1 / sizeof(T);
+            const npy_intp sdst  = b_sdst  / sizeof(T);
+
+            HWY_LANES_CONSTEXPR int vstep = Lanes<T>();
+            const int wstep = vstep * 2;
+            const int hstep = vstep / 2;
+
+            // lots of specializations, to squeeze out max performance
+            // contiguous
+            if (ssrc0 == 2 && ssrc0 == ssrc1 && ssrc0 == sdst) {
+                for (; len >= vstep; len -= vstep, src0 += wstep, src1 += wstep, dst += wstep) {
+                    auto a0 = LoadU<T>(src0);
```

Comment:
```suggestion
                    V a0 = LoadU(src0);
```
Avoid auto. `LoadU` should detect the pointer type, so no need for explicit deduction.